cmd/zfs/zfs_main.c

```diff
@@ -1265,7 +1276,8 @@ zfs_do_create(int argc, char **argv)
 		log_history = B_FALSE;
 	}
 
-	ret = zfs_mount_and_share(g_zfs, argv[0], ZFS_TYPE_DATASET);
+	if (!nomount)
+		ret = zfs_mount_and_share(g_zfs, argv[0], ZFS_TYPE_DATASET);
```

Comment:
Skipping the `zfs_mount_and_share` here looks like it will incorrectly leave the return value `ret` as `1`.